audbcards/core/datacard.py

```diff
@@ -161,83 +147,143 @@ def file_duration_distribution(self) -> str:
-                :attr:`audbcards.Datacard.example_media`
-                is used
+        If :attr:`audbcards.Datacard.sphinx_build_dir`
+        or :attr:`audbcards.Datacard.sphinx_src_dir`
+        is not ``None``,
+        an example media file is cached in the folder
+        ``<dataset-name>-<dataset-version>-player-media/``
+        inside
+        ``<cache-root>/<dataset-name>/<dataset-version>/``,
+        using the same sub-folder structure
+        as the media file has inside its dataset.
+        If :attr:`audbcards.Datacard.sphinx_build_dir`
+        is not ``None``,
+        the media sub-folder structure
+        is also copied
+        to the sphinx build dir into
+        ``<sphinx-build-dir>/<path>/<dataset-name>/``.
+
+        If :attr:`audbcards.Datacard.sphinx_src_dir` is not ``None``,
+        a plot of the waveform of the media file
+        is cached under
+        ``<dataset-name>-<dataset-version>-player-waveform.png``
+        inside
+        ``<cache-root>/<dataset-name>/<dataset-version>/``.
+        It is also copied to the sphinx source folder into
+        ``<sphinx-src-dir>/<path>/<dataset-name>/``.
 
-        """
-        if file is None:
-            file = self.example_media
+        Returns:
+            String containing RST code to include the player
 
-        # use audb cache instead of dataset.cache_root
-        media_src_dir = (
-            f"{audb.default_cache_root()}/"
-            f"{audb.flavor_path(self.dataset.name, self.dataset.version)}"
+        """
+        # Cache is organized as `<cache_root>/<name>/<version>/`
+        cache_folder = audeer.path(
+            self.cache_root,
+            self.dataset.name,
+            self.dataset.version,
+        )
+        cache_example_media = audeer.path(
+            cache_folder,
+            f"{self.dataset.name}-{self.dataset.version}-player-media",
+            self.dataset.example_media,
+        )
+        cache_waveform_file = audeer.path(
+            cache_folder,
+            f"{self.dataset.name}-{self.dataset.version}-player-waveform.png",
         )
 
-        # Move file to build folder
-        if self.sphinx_build_dir is not None:
-            media_dst_dir = audeer.path(
-                self.sphinx_build_dir,
+        def load_media_to_cache(cache_example_media: str):
+            r"""Load media file with audb and copy to audbcards cache.
+
+            Args:
+                cache_example_media: full path to media file in cache
+
+            """
+            media = audb.load_media(
+                self.dataset.name,
+                self.dataset.example_media,
+                version=self.dataset.version,
+                verbose=False,
+            )[0]
+            audeer.mkdir(os.path.dirname(cache_example_media))
+            shutil.copy(media, cache_example_media)
+
+        # Add plot of waveform
+        if self.sphinx_src_dir is not None:
+            if not os.path.exists(cache_example_media):
```

Comment:
Would it even make sense to handle this conditional in `load_media_to_cache` , together with the defintion of `cache_example_media` from the comment above? I mean the handling of `cache_example_media` is only relevant for the task performed in `load_media_to_cache`.

Reply:
Could idea, I added the handling of the cache to the `load_media_to_cache()` and the new `plot_waveform_to_cache()` functions, that now simply return the path to the file in cache, as we need to use those values afterwards.

Reply:
This is done.